### DISCUSSION TITLE
GitHub Flavored Markdown doesn't render CSS styles inside a HTML block

### DISCUSSION THREAD

**Discussion by ZeroAurora** [community/community#22728] [New to GitHub]
<p>Hey here,<br>
I’m recently writing a README.md and I want to decorate it with styles available in CSS, <code>text-align: center</code> for example, since the <code>&lt;center&gt;</code> tag and the <code>align</code> attribute is deprecated as said in MDN. So I used a HTML block in the markdown file:</p>
<pre><code class="lang-markdown">&lt;div style="text-align: center;"&gt;

*Here are some markdowns.*

&lt;/div&gt;
</code></pre>
<p>However when GitHub rendered it, the <code>style</code> attribute was automatically removed and no CSS style was applied.</p>
<pre><code class="lang-auto">&lt;div&gt;

&lt;p&gt;&lt;i&gt;Here are some markdowns.&lt;/i&gt;&lt;/p&gt;

&lt;/div&gt;
</code></pre>
<p>I wonder if there is a way to use CSS within a markdown file. If not, I’d like to know why. Thank you!</p>

**Comment by SethClydesdale** [comment, marked as answer]
<p>Hello <a class="mention" href="/zeroaurora">@ZeroAurora</a>,</p>
<p>Unfortunately you cannot use CSS in GitHub markdown as it is a part of the sanitization process.</p>
<blockquote>
<p>The HTML is sanitized, aggressively removing things that could harm you and your kin—such as  <code>script</code>  tags, inline-styles, and  <code>class</code>  or  <code>id</code>  attributes.</p>
</blockquote>
<p><em>source: <a href="https://github.com/github/markup" rel="nofollow noopener">https://github.com/github/markup</a></em></p>
<p>While CSS can’t be super malicious, one could create a number of nuisances with it, such as for example, an element that takes up the entire page and blocks user interaction. I admit it would be nice to use at least some styles for alignment and what not, but it’s understandable.</p>
<p>In regards to centering, the <code>align</code> attribute is the only method that will work currently. Despite being deprecated, it’s still rendered, for now anyway.</p>
<pre><code class="lang-auto">&lt;p align="center"&gt;centered content&lt;/p&gt;
</code></pre>

**Reply by bilogic** [reply]
Can it be bypassed?

**Reply by JaiS99** [reply]
no

**Reply by levdoescode** [reply]
> Can it be bypassed?

You can embed styling inside an SVG
https://pragmaticpineapple.com/adding-custom-html-and-css-to-github-readme/

**Reply by scottyschup** [reply]
The SVG hack doesn't appear to work anymore :/

**Reply by LizzieSpace** [reply]
let's hope the `allign` atribute continues rendering for a veery long time...
